src/normi/_estimators.py

```diff
@@ -131,21 +131,26 @@ def fit(
 
         """
         self._reset()
-
         _check_X(X=X, n_dims=self.n_dims)
 
         # define number of features and samples
         n_samples: int
         n_cols: int
         n_samples, n_cols = X.shape
         self._n_samples: int = n_samples
-        self._n_features: int = n_cols // self.n_dims
+        if isinstance(self.n_dims, int):
+            self._n_features: int = n_cols // self.n_dims
+        else:
+            self._n_features: int = len(self.n_dims)
 
         # scale input
         X = StandardScaler().fit_transform(X)
-        X = np.split(X, self._n_features, axis=1)
+        if isinstance(self.n_dims, int):
+            X = np.split(X, self._n_features, axis=1)
+        else:
+            X = np.split(X, np.cumsum(self.n_dims), axis=1)[:-1]
 
-        self.mi_: PositiveMatrix
+        self.mi_: FloatMatrix
```

Comment:
Did you face an issue similar to #6 or #7? In theory, the mutual information should be positive. But I guess the same holds for the entropy, so this would solve raising ugly beartype errors.

Reply:
Yes. I am suffering negative MIs and entropies. This enables to get _some results even when there are some numerical problems leading to negative values.